### PR TITLE
Revert digest usage for tenant-registry images

### PR DESCRIPTION
**Pull request by 0sewa0**
As the tenant-registry is not made to be used with digest, this change goes back to the previous solution of using the build-label on the images of the tenant-registry in order to determine the version, so we can update the oneagent/activegate when necessary.

## How can this be tested?
deploy oneagents/activegates
check if the `image` field only uses the tag, and check that the version label on the statefulset and daemonset is set.

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/controllers/dynakube/dynakube_controller.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func NewDynaKubeController(kubeClient client.Client, apiReader client.Reader, sc
 		config:                 config,
 		operatorNamespace:      os.Getenv(kubeobjects.EnvPodNamespace),
 		clusterID:              clusterID,
-		digestProvider:         version.GetImageDigest,
+		digestProvider:         version.GetImageVersion,
```
Comment (luhi-DT) [inline src/controllers/dynakube/dynakube_controller.go:70]: should be renamed to versionProvider i guess

File: src/controllers/dynakube/version/activegate.go [hunk 2]
```diff
@@ -11,13 +11,13 @@ import (
 type activeGateUpdater struct {
 	dynakube   *dynatracev1beta1.DynaKube
 	dtClient   dtclient.Client
-	digestFunc ImageDigestFunc
+	digestFunc ImageVersionFunc
```
Comment (luhi-DT) [inline src/controllers/dynakube/version/activegate.go:14]: should be renamed to versionFunc i guess

File: src/controllers/dynakube/version/updater.go [hunk 3]
```diff
@@ -139,6 +140,39 @@ func updateVersionStatus(
...
+
+	log.Info("updating image version info for tenant registry image",
+		"image", imageUri,
+		"oldImageID", target.ImageID,
+		"oldVersion", target.Version)
+
+	if taggedRef, ok := ref.(reference.NamedTagged); ok {
+		imageVersion, err := imageVersionFunc(ctx, imageUri, dockerCfg)
+		if err != nil {
+			log.Error(err, "failed to get image digest, falling back to tag")
```
Comment (luhi-DT) [inline src/controllers/dynakube/version/updater.go:163]: This log is wrong as you actually fail to get the image version and not the image digest

**Review by luhi-DT — DISMISSED** [review]
check comments from before

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**